kns

```diff
@@ -13,12 +13,13 @@ if [ ! -x "$(which fzf 2>/dev/null)" ]; then
   exit 1
 fi
 
-current="$(kubectl config current-context)"
-namespace="$(kubectl config get-contexts "$current" | awk "/$current/ {print \$5}")"
+export current="$(kubectl config current-context)"
+namespace="$(kubectl config view -o jsonpath="{.contexts[?(@.name == '${current}')].context.namespace}")"
 if [ -z "$namespace" ]; then
   namespace="default"
 fi
-selected=$( (kubectl get namespaces -o name | sed "s-namespaces/--"; echo $namespace ) | fzf -0 -1 --tac -q "${1:-""}" --prompt "$current> " --preview "kubectl --namespace {} get pods")
+export namespace
```

Comment:
I might have to safe the original namespace environment variable here to not cause the script to mess up the calling environment. But it did not find out how to properly do the jsonpath extraction without a shell variable.